auditing/timescaledb.go

```diff
@@ -0,0 +1,345 @@
+		Password string
+
+		// Retention defines when audit traces will be thrown away, only settable on initial database usage
+		// If this needs to be changed over time, you need to do this manually. Defaults to '14 days'.
+		Retention string
+		// CompressionInterval defines after which period audit traces will be compressed, only settable on initial database usage.
+		// If this needs to be changed over time, you need to do this manually. Defaults to '7 days'.
+		CompressionInterval string
+		// ChunkInterval defines after which period audit traces will be stored in a new chunk table, only settable on initial database usage.
+		// If this needs to be changed over time, you need to do this manually. Defaults to '1 days'.
+		ChunkInterval string
+	}
+
+	timescaleAuditing struct {
+		component string
+		db        *sqlx.DB
+		log       *slog.Logger
+
+		config *TimescaleDbConfig
+	}
+
+	timescaledbRow struct {
+		Timestamp time.Time `db:"timestamp"`
+		Entry     []byte    `db:"entry"`
+	}
+
+	sqlCompOp string
+)
+
+const (
+	equals sqlCompOp = "equals"
+	like   sqlCompOp = "like"
+	phrase sqlCompOp = "phrase"
+)
+
+func NewTimescaleDB(c Config, tc TimescaleDbConfig) (Auditing, error) {
+	if c.Component == "" {
+		component, err := defaultComponent()
+		if err != nil {
+			return nil, err
+		}
+
+		c.Component = component
+	}
+
+	if tc.Port == "" {
+		tc.Port = "5432"
+	}
+	if tc.DB == "" {
+		tc.DB = "postgres"
+	}
+	if tc.User == "" {
+		tc.User = "postgres"
+	}
+	if tc.Retention == "" {
+		tc.Retention = "14 days"
+	}
+	if tc.ChunkInterval == "" {
+		tc.ChunkInterval = "1 days"
+	}
+	if tc.CompressionInterval == "" {
+		tc.CompressionInterval = "7 days"
+	}
+
+	source := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=disable", tc.Host, tc.Port, tc.User, tc.DB, tc.Password)
+
+	db, err := sqlx.Connect("postgres", source)
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to datastore: %w", err)
+	}
+
+	a := &timescaleAuditing{
+		component: c.Component,
+		log:       c.Log.WithGroup("auditing"),
+		db:        db,
+		config:    &tc,
+	}
+
+	err = a.initialize()
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize timescaledb backend: %w", err)
+	}
+
+	a.log.Info("connected to timescaledb backend")
+
+	return a, nil
+}
+
+func (a *timescaleAuditing) initialize() error {
+	type txStatement struct {
+		query string
+		args  []any
+	}
+
+	initialSchema := &migrator.Migration{
+		Name: "Initial database schema",
+		Func: func(tx *sql.Tx) error {
+			for _, stmt := range []txStatement{
+				{
+					query: `CREATE EXTENSION IF NOT EXISTS timescaledb`,
+				},
+				{
+					query: `CREATE EXTENSION IF NOT EXISTS pg_stat_statements`,
+				},
+				{
+					query: `CREATE TABLE IF NOT EXISTS traces (
+						timestamp timestamp NOT NULL,
+						entry jsonb NOT NULL
+					)`,
+				},
+				{
+					query: `ALTER TABLE traces ADD COLUMN ts tsvector GENERATED ALWAYS AS (to_tsvector('simple', entry)) STORED;
+`,
+				},
```

Comment:
Why not adding this column during `create table` which would be much more readable